Sources/NextcloudFileProviderKit/Utilities/LocalFiles.swift

```diff
@@ -51,23 +51,49 @@ public func isLockFileName(_ filename: String) -> Bool {
 ///
 /// Example for Microsoft Office: `MyDoc.docx` is extracted from `~$MyDoc.docx`.
 /// Example for LibreOffice: `MyDoc.odt` is extracted from `.~lock.MyDoc.odt#`.
-/// 
+/// Filename with <8 characters: Test.docx → lock file: ~$Test.docx
+/// Filename with >8 characters: Document.docx → lock file: ~$cument.docx
+/// Filename sandbox-style temporary naming like: Welcome123456.doc.sb-d215eb53-IBAwfU
 /// - Returns: Either the original file name parsed from the given lock file name or `nil`, if it is not a recognized lock file format.
 ///
-public func originalFileName(fromLockFileName lockFilename: String) -> String? {
+public func originalFileName(fromLockFileName lockFilename: String, dbManager: FilesDatabaseManager) -> String? {
+    let logger = FileProviderLogger(category: "localfileutils", log: dbManager.logger.log)
```

Comment:
Also not a blocker, just an explanation how I usually define the `category` argument: Named like the type the `FileProviderLogger` is setup for. If it is not in scope of a type but a loose function in a file (as here), it is best to name it like the file so it is easy to associate the origin of log messages.

```suggestion
    let logger = FileProviderLogger(category: "LocalFiles", log: dbManager.logger.log)
```